src/CodeGen_Hexagon.cpp

```diff
@@ -1319,6 +1319,123 @@ void CodeGen_Hexagon::visit(const Cast *op) {
     CodeGen_Posix::visit(op);
 }
 
+void CodeGen_Hexagon::codegen_predicated_vector_load(const Call *op) {
+    const Call *load_addr = op->args[0].as<Call>();
+    internal_assert(load_addr && (load_addr->is_intrinsic(Call::address_of)))
+        << "The first argument to predicated_load must be call to address_of of the load\n";
+    const Broadcast *broadcast = load_addr->args[0].as<Broadcast>();
+    const Load *load = broadcast ? broadcast->value.as<Load>() : load_addr->args[0].as<Load>();
+    internal_assert(load) << "The sole argument to address_of must be a load or broadcast of load\n";
+
+    // If it's a Handle, load it as a uint64_t and then cast
+    if (load->type.is_handle()) {
+        Expr uint64_load = Load::make(UInt(64, load->type.lanes()), load->name, load->index, load->image, load->param);
+        Expr src = Call::make(Handle().with_lanes(uint64_load.type().lanes()), Call::address_of, {uint64_load}, Call::Intrinsic);
+        Expr expr = Call::make(uint64_load.type(), Call::predicated_load, {src, op->args[1]}, Call::Intrinsic);
+        codegen(reinterpret(load->type, expr));
+        return;
+    }
+
+    // We need to scalarize the predicated store since masked load on
+    // hexagon is not handled by the LLVM
+
+    // If any of the predicate lane is true, then just do normal load; otherwise,
```

Comment:
Might be worth a comment that this relies on the fact that Hexagon allocations have one vector past the end allocated.